pkg/gator/expand.go

```diff
@@ -0,0 +1,228 @@
+			return nil, err
+		}
+	}
+
+	var resultants []*unstructured.Unstructured
+	for _, gen := range er.objects {
+		ns, err := er.namespaceForGenerator(gen)
+		base := &types.Mutable{
+			Object:    gen,
+			Namespace: ns,
+			Username:  "",
+			Source:    types.SourceTypeGenerated,
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error expanding generator: %s", err)
+		}
+		r, err := expSystem.Expand(base)
+		if err != nil {
+			return nil, fmt.Errorf("error expanding generator: %s", err)
+		}
+		resultants = append(resultants, r...)
+	}
+
+	return resultants, nil
+}
+
+func (er *expansionResources) namespaceForGenerator(gen *unstructured.Unstructured) (*corev1.Namespace, error) {
+	genNs := gen.GetNamespace()
+	if genNs == "" {
+		return &corev1.Namespace{}, nil
+	}
+
+	ns, exists := er.namespaces[genNs]
+	if !exists {
+		if genNs == "default" {
+			return &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}, nil
+		}
+		return nil, fmt.Errorf("namespace resource %q not found in supplied configs", genNs)
+	}
+	return ns, nil
+}
+
+func (er *expansionResources) addResources(resources []*unstructured.Unstructured) error {
+	for _, r := range resources {
+		if err := er.add(r); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (er *expansionResources) addMutator(mut *unstructured.Unstructured) error {
+	var mutErr error
+	var m types.Mutator
+
+	switch mut.GetKind() {
+	case "Assign":
+		a, err := convertAssign(mut)
+		if err != nil {
+			return err
+		}
+		m, mutErr = assign.MutatorForAssign(a)
+	case "AssignMetadata":
+		a, err := convertAssignMetadata(mut)
+		if err != nil {
+			return err
+		}
+		m, mutErr = assignmeta.MutatorForAssignMetadata(a)
+	case "ModifySet":
+		ms, err := convertModifySet(mut)
+		if err != nil {
+			return err
+		}
+		m, mutErr = modifyset.MutatorForModifySet(ms)
+	default:
+		return fmt.Errorf("cannot convert mutator of kind %q", mut.GetKind())
+	}
+
+	if mutErr != nil {
+		return mutErr
+	}
+	er.mutators = append(er.mutators, m)
+	return nil
+}
+
+func (er *expansionResources) add(u *unstructured.Unstructured) error {
+	var err error
+	switch {
+	case isMutator(u):
+		err = er.addMutator(u)
+	case isExpansion(u):
+		err = er.addExpansionTemplate(u)
+	case isNamespace(u):
+		err = er.addNamespace(u)
+	}
+
+	if err != nil {
+		// Any resource can technically be a generator
+		er.objects = append(er.objects, u)
+	}
+
+	return err
+}
+
+func (er *expansionResources) addExpansionTemplate(u *unstructured.Unstructured) error {
+	te, err := convertExpansionTemplate(u)
+	if err != nil {
+		return err
+	}
+	er.templateExpansions = append(er.templateExpansions, te)
+	return nil
+}
+
+func (er *expansionResources) addNamespace(u *unstructured.Unstructured) error {
+	ns, err := convertNamespace(u)
+	if err != nil {
+		return err
+	}
+	er.namespaces[ns.GetName()] = ns
+	return nil
+}
+
+func isExpansion(u *unstructured.Unstructured) bool {
+	_, exists := ExpansionAPIVersions[u.GetAPIVersion()]
```

Comment:
Not relying on the version, just the group, will make it easier to add more versions in the future and help with cross-k8s-version compatibility for things like Namespaces